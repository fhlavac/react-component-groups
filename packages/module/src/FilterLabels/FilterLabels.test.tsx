import React from 'react';
import { render } from '@testing-library/react';
import FilterLabels, { FilterLabelsProps } from './FilterLabels';

describe('FilterLabels component', () => {
  const filters: FilterLabelsProps['filters'] = [
    { text: 'Label 1', count: 5 },
    { text: 'Label 2', count: 2 },
    {
      categoryName: 'Group 1',
      labels: [ { text: 'Label 3' }, { text: 'Label 4', count: 3 } ],
    },
  ];

  it('should render FilterLabels component with default props', () => {
    const { container } = render(
      <FilterLabels 
        filters={filters} 
        onDelete={jest.fn()} 
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render FilterLabels component with custom delete all button title', () => {
    const { container } = render(
      <FilterLabels 
        filters={filters} 
        onDelete={jest.fn()} 
        deleteAllButtonTitle="Remove all filters"
        showDeleteAllButton={true}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render FilterLabels component with delete group button enabled', () => {
    const { container } = render(
      <FilterLabels 
        filters={filters} 
        onDelete={jest.fn()} 
        showDeleteGroupButton={true}
      />
    );
    expect(container).toMatchSnapshot();
  });
});