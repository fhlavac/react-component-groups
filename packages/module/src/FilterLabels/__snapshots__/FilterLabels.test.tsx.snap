// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterLabels component should render FilterLabels component with custom delete all button title 1`] = `
<div>
  <span
    class="chipFilters-0-2-1"
  >
    <div
      aria-labelledby="pf-random-id-7"
      class="pf-v5-c-chip-group pf-m-category"
      data-ouia-component-id="FilterLabels-group-Group 1"
      data-ouia-component-type="PF5/ChipGroup"
      data-ouia-safe="true"
      role="group"
    >
      <div
        class="pf-v5-c-chip-group__main"
      >
        <span
          class="pf-v5-c-chip-group__label"
          id="pf-random-id-7"
        >
          Group 1
        </span>
        <ul
          aria-labelledby="pf-random-id-7"
          class="pf-v5-c-chip-group__list"
          role="list"
        >
          <li
            class="pf-v5-c-chip-group__list-item"
          >
            <div
              class="pf-v5-c-chip"
              data-ouia-component-id="FilterLabels-label-Group 1-Label 3"
              data-ouia-component-type="PF5/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v5-c-chip__content"
              >
                <span
                  class="pf-v5-c-chip__text"
                  id="pf-random-id-8"
                >
                  Label 3
                </span>
              </span>
              <span
                class="pf-v5-c-chip__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-8 pf-random-id-8"
                  class="pf-v5-c-button pf-m-plain"
                  data-ouia-component-id="FilterLabels-label-Group 1-Label 3"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe="true"
                  id="remove_pf-random-id-8"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </li>
          <li
            class="pf-v5-c-chip-group__list-item"
          >
            <div
              class="pf-v5-c-chip"
              data-ouia-component-id="FilterLabels-label-Group 1-Label 4"
              data-ouia-component-type="PF5/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v5-c-chip__content"
              >
                <span
                  class="pf-v5-c-chip__text"
                  id="pf-random-id-9"
                >
                  Label 4
                  <span
                    class="pf-v5-c-badge pf-m-unread"
                    data-ouia-component-id="FilterLabels-label-Group 1-Label 4-badge"
                  >
                    3
                  </span>
                </span>
              </span>
              <span
                class="pf-v5-c-chip__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-9 pf-random-id-9"
                  class="pf-v5-c-button pf-m-plain"
                  data-ouia-component-id="FilterLabels-label-Group 1-Label 4"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe="true"
                  id="remove_pf-random-id-9"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      aria-label="Chip group category"
      class="pf-v5-c-chip-group"
      data-ouia-component-id="FilterLabels-group-Label 1"
      data-ouia-component-type="PF5/ChipGroup"
      data-ouia-safe="true"
      role="group"
    >
      <div
        class="pf-v5-c-chip-group__main"
      >
        <ul
          aria-label="Chip group category"
          class="pf-v5-c-chip-group__list"
          role="list"
        >
          <li
            class="pf-v5-c-chip-group__list-item"
          >
            <div
              class="pf-v5-c-chip"
              data-ouia-component-id="FilterLabels-label-Label 1"
              data-ouia-component-type="PF5/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v5-c-chip__content"
              >
                <span
                  class="pf-v5-c-chip__text"
                  id="pf-random-id-11"
                >
                  Label 1
                  <span
                    class="pf-v5-c-badge pf-m-unread"
                    data-ouia-component-id="FilterLabels-label-Label 1-badge"
                  >
                    5
                  </span>
                </span>
              </span>
              <span
                class="pf-v5-c-chip__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-11 pf-random-id-11"
                  class="pf-v5-c-button pf-m-plain"
                  data-ouia-component-id="FilterLabels-label-Label 1"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe="true"
                  id="remove_pf-random-id-11"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      aria-label="Chip group category"
      class="pf-v5-c-chip-group"
      data-ouia-component-id="FilterLabels-group-Label 2"
      data-ouia-component-type="PF5/ChipGroup"
      data-ouia-safe="true"
      role="group"
    >
      <div
        class="pf-v5-c-chip-group__main"
      >
        <ul
          aria-label="Chip group category"
          class="pf-v5-c-chip-group__list"
          role="list"
        >
          <li
            class="pf-v5-c-chip-group__list-item"
          >
            <div
              class="pf-v5-c-chip"
              data-ouia-component-id="FilterLabels-label-Label 2"
              data-ouia-component-type="PF5/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v5-c-chip__content"
              >
                <span
                  class="pf-v5-c-chip__text"
                  id="pf-random-id-13"
                >
                  Label 2
                  <span
                    class="pf-v5-c-badge pf-m-unread"
                    data-ouia-component-id="FilterLabels-label-Label 2-badge"
                  >
                    2
                  </span>
                </span>
              </span>
              <span
                class="pf-v5-c-chip__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-13 pf-random-id-13"
                  class="pf-v5-c-button pf-m-plain"
                  data-ouia-component-id="FilterLabels-label-Label 2"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe="true"
                  id="remove_pf-random-id-13"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <button
      aria-disabled="false"
      class="pf-v5-c-button pf-m-link"
      data-ouia-component-id="FilterLabels-delete-all-button"
      data-ouia-component-type="PF5/Button"
      data-ouia-safe="true"
      type="button"
    >
      Remove all filters
    </button>
  </span>
</div>
`;

exports[`FilterLabels component should render FilterLabels component with default props 1`] = `
<div>
  <span
    class="chipFilters-0-2-1"
  >
    <div
      aria-labelledby="pf-random-id-0"
      class="pf-v5-c-chip-group pf-m-category"
      data-ouia-component-id="FilterLabels-group-Group 1"
      data-ouia-component-type="PF5/ChipGroup"
      data-ouia-safe="true"
      role="group"
    >
      <div
        class="pf-v5-c-chip-group__main"
      >
        <span
          class="pf-v5-c-chip-group__label"
          id="pf-random-id-0"
        >
          Group 1
        </span>
        <ul
          aria-labelledby="pf-random-id-0"
          class="pf-v5-c-chip-group__list"
          role="list"
        >
          <li
            class="pf-v5-c-chip-group__list-item"
          >
            <div
              class="pf-v5-c-chip"
              data-ouia-component-id="FilterLabels-label-Group 1-Label 3"
              data-ouia-component-type="PF5/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v5-c-chip__content"
              >
                <span
                  class="pf-v5-c-chip__text"
                  id="pf-random-id-1"
                >
                  Label 3
                </span>
              </span>
              <span
                class="pf-v5-c-chip__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
                  class="pf-v5-c-button pf-m-plain"
                  data-ouia-component-id="FilterLabels-label-Group 1-Label 3"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe="true"
                  id="remove_pf-random-id-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </li>
          <li
            class="pf-v5-c-chip-group__list-item"
          >
            <div
              class="pf-v5-c-chip"
              data-ouia-component-id="FilterLabels-label-Group 1-Label 4"
              data-ouia-component-type="PF5/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v5-c-chip__content"
              >
                <span
                  class="pf-v5-c-chip__text"
                  id="pf-random-id-2"
                >
                  Label 4
                  <span
                    class="pf-v5-c-badge pf-m-unread"
                    data-ouia-component-id="FilterLabels-label-Group 1-Label 4-badge"
                  >
                    3
                  </span>
                </span>
              </span>
              <span
                class="pf-v5-c-chip__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-2 pf-random-id-2"
                  class="pf-v5-c-button pf-m-plain"
                  data-ouia-component-id="FilterLabels-label-Group 1-Label 4"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe="true"
                  id="remove_pf-random-id-2"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      aria-label="Chip group category"
      class="pf-v5-c-chip-group"
      data-ouia-component-id="FilterLabels-group-Label 1"
      data-ouia-component-type="PF5/ChipGroup"
      data-ouia-safe="true"
      role="group"
    >
      <div
        class="pf-v5-c-chip-group__main"
      >
        <ul
          aria-label="Chip group category"
          class="pf-v5-c-chip-group__list"
          role="list"
        >
          <li
            class="pf-v5-c-chip-group__list-item"
          >
            <div
              class="pf-v5-c-chip"
              data-ouia-component-id="FilterLabels-label-Label 1"
              data-ouia-component-type="PF5/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v5-c-chip__content"
              >
                <span
                  class="pf-v5-c-chip__text"
                  id="pf-random-id-4"
                >
                  Label 1
                  <span
                    class="pf-v5-c-badge pf-m-unread"
                    data-ouia-component-id="FilterLabels-label-Label 1-badge"
                  >
                    5
                  </span>
                </span>
              </span>
              <span
                class="pf-v5-c-chip__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-4 pf-random-id-4"
                  class="pf-v5-c-button pf-m-plain"
                  data-ouia-component-id="FilterLabels-label-Label 1"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe="true"
                  id="remove_pf-random-id-4"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      aria-label="Chip group category"
      class="pf-v5-c-chip-group"
      data-ouia-component-id="FilterLabels-group-Label 2"
      data-ouia-component-type="PF5/ChipGroup"
      data-ouia-safe="true"
      role="group"
    >
      <div
        class="pf-v5-c-chip-group__main"
      >
        <ul
          aria-label="Chip group category"
          class="pf-v5-c-chip-group__list"
          role="list"
        >
          <li
            class="pf-v5-c-chip-group__list-item"
          >
            <div
              class="pf-v5-c-chip"
              data-ouia-component-id="FilterLabels-label-Label 2"
              data-ouia-component-type="PF5/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v5-c-chip__content"
              >
                <span
                  class="pf-v5-c-chip__text"
                  id="pf-random-id-6"
                >
                  Label 2
                  <span
                    class="pf-v5-c-badge pf-m-unread"
                    data-ouia-component-id="FilterLabels-label-Label 2-badge"
                  >
                    2
                  </span>
                </span>
              </span>
              <span
                class="pf-v5-c-chip__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-6 pf-random-id-6"
                  class="pf-v5-c-button pf-m-plain"
                  data-ouia-component-id="FilterLabels-label-Label 2"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe="true"
                  id="remove_pf-random-id-6"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <button
      aria-disabled="false"
      class="pf-v5-c-button pf-m-link"
      data-ouia-component-id="FilterLabels-delete-all-button"
      data-ouia-component-type="PF5/Button"
      data-ouia-safe="true"
      type="button"
    >
      Clear filters
    </button>
  </span>
</div>
`;

exports[`FilterLabels component should render FilterLabels component with delete group button enabled 1`] = `
<div>
  <span
    class="chipFilters-0-2-1"
  >
    <div
      aria-labelledby="pf-random-id-14"
      class="pf-v5-c-chip-group pf-m-category"
      data-ouia-component-id="FilterLabels-group-Group 1"
      data-ouia-component-type="PF5/ChipGroup"
      data-ouia-safe="true"
      role="group"
    >
      <div
        class="pf-v5-c-chip-group__main"
      >
        <span
          class="pf-v5-c-chip-group__label"
          id="pf-random-id-14"
        >
          Group 1
        </span>
        <ul
          aria-labelledby="pf-random-id-14"
          class="pf-v5-c-chip-group__list"
          role="list"
        >
          <li
            class="pf-v5-c-chip-group__list-item"
          >
            <div
              class="pf-v5-c-chip"
              data-ouia-component-id="FilterLabels-label-Group 1-Label 3"
              data-ouia-component-type="PF5/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v5-c-chip__content"
              >
                <span
                  class="pf-v5-c-chip__text"
                  id="pf-random-id-15"
                >
                  Label 3
                </span>
              </span>
              <span
                class="pf-v5-c-chip__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-15 pf-random-id-15"
                  class="pf-v5-c-button pf-m-plain"
                  data-ouia-component-id="FilterLabels-label-Group 1-Label 3"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe="true"
                  id="remove_pf-random-id-15"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </li>
          <li
            class="pf-v5-c-chip-group__list-item"
          >
            <div
              class="pf-v5-c-chip"
              data-ouia-component-id="FilterLabels-label-Group 1-Label 4"
              data-ouia-component-type="PF5/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v5-c-chip__content"
              >
                <span
                  class="pf-v5-c-chip__text"
                  id="pf-random-id-16"
                >
                  Label 4
                  <span
                    class="pf-v5-c-badge pf-m-unread"
                    data-ouia-component-id="FilterLabels-label-Group 1-Label 4-badge"
                  >
                    3
                  </span>
                </span>
              </span>
              <span
                class="pf-v5-c-chip__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-16 pf-random-id-16"
                  class="pf-v5-c-button pf-m-plain"
                  data-ouia-component-id="FilterLabels-label-Group 1-Label 4"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe="true"
                  id="remove_pf-random-id-16"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="pf-v5-c-chip-group__close"
      >
        <button
          aria-disabled="false"
          aria-label="Close chip group"
          aria-labelledby="remove_group_pf-random-id-14 pf-random-id-14"
          class="pf-v5-c-button pf-m-plain"
          data-ouia-component-id="FilterLabels-group-Group 1"
          data-ouia-component-type="PF5/Button"
          data-ouia-safe="true"
          id="remove_group_pf-random-id-14"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      aria-label="Chip group category"
      class="pf-v5-c-chip-group"
      data-ouia-component-id="FilterLabels-group-Label 1"
      data-ouia-component-type="PF5/ChipGroup"
      data-ouia-safe="true"
      role="group"
    >
      <div
        class="pf-v5-c-chip-group__main"
      >
        <ul
          aria-label="Chip group category"
          class="pf-v5-c-chip-group__list"
          role="list"
        >
          <li
            class="pf-v5-c-chip-group__list-item"
          >
            <div
              class="pf-v5-c-chip"
              data-ouia-component-id="FilterLabels-label-Label 1"
              data-ouia-component-type="PF5/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v5-c-chip__content"
              >
                <span
                  class="pf-v5-c-chip__text"
                  id="pf-random-id-18"
                >
                  Label 1
                  <span
                    class="pf-v5-c-badge pf-m-unread"
                    data-ouia-component-id="FilterLabels-label-Label 1-badge"
                  >
                    5
                  </span>
                </span>
              </span>
              <span
                class="pf-v5-c-chip__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-18 pf-random-id-18"
                  class="pf-v5-c-button pf-m-plain"
                  data-ouia-component-id="FilterLabels-label-Label 1"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe="true"
                  id="remove_pf-random-id-18"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      aria-label="Chip group category"
      class="pf-v5-c-chip-group"
      data-ouia-component-id="FilterLabels-group-Label 2"
      data-ouia-component-type="PF5/ChipGroup"
      data-ouia-safe="true"
      role="group"
    >
      <div
        class="pf-v5-c-chip-group__main"
      >
        <ul
          aria-label="Chip group category"
          class="pf-v5-c-chip-group__list"
          role="list"
        >
          <li
            class="pf-v5-c-chip-group__list-item"
          >
            <div
              class="pf-v5-c-chip"
              data-ouia-component-id="FilterLabels-label-Label 2"
              data-ouia-component-type="PF5/Chip"
              data-ouia-safe="true"
            >
              <span
                class="pf-v5-c-chip__content"
              >
                <span
                  class="pf-v5-c-chip__text"
                  id="pf-random-id-20"
                >
                  Label 2
                  <span
                    class="pf-v5-c-badge pf-m-unread"
                    data-ouia-component-id="FilterLabels-label-Label 2-badge"
                  >
                    2
                  </span>
                </span>
              </span>
              <span
                class="pf-v5-c-chip__actions"
              >
                <button
                  aria-disabled="false"
                  aria-label="close"
                  aria-labelledby="remove_pf-random-id-20 pf-random-id-20"
                  class="pf-v5-c-button pf-m-plain"
                  data-ouia-component-id="FilterLabels-label-Label 2"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe="true"
                  id="remove_pf-random-id-20"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <button
      aria-disabled="false"
      class="pf-v5-c-button pf-m-link"
      data-ouia-component-id="FilterLabels-delete-all-button"
      data-ouia-component-type="PF5/Button"
      data-ouia-safe="true"
      type="button"
    >
      Clear filters
    </button>
  </span>
</div>
`;
